src/oscar/apps/order/utils.py

```diff
@@ -119,6 +118,8 @@ def create_order_model(self, user, basket, shipping_address,
             order_data['status'] = status
         if extra_order_fields:
             order_data.update(extra_order_fields)
+        if 'site' not in order_data:
+            order_data['site'] = Site._default_manager.get_current()
```

Comment:
Could be replaced with `order_data.setdefault('site', Site._default_manager.get_current())`
Original idea by @mvantellingen :wink: